libosmscout-gpx/src/osmscout/gpx/Import.cpp

```diff
@@ -718,6 +718,37 @@ class RouteContext : public GpxParserContext {
   }
 };
 
+class TrkExtensionContext : public GpxParserContext {
+private:
+  Track &track;
+public:
+  TrkExtensionContext(xmlParserCtxtPtr ctxt, Track &track, GpxParser &parser) :
+      GpxParserContext(ctxt, parser), track(track) {}
+
+  ~TrkExtensionContext() override
+  {
+  }
+
+  const char *ContextName() const override
+  {
+    return "TrkExtensions";
+  }
+
+  GpxParserContext* StartElement(const std::string &name,
+                                 const std::unordered_map<std::string, std::string> &/*atts*/) override
+  {
+    if (name=="gpxx:TrackExtension"){
+      return new TrkExtensionContext(ctxt, track, parser);
+    } else if (name=="gpxx:DisplayColor"){
```

Comment:
I am worry that it would need to be more complicated. Because you can name extension namespace differently than `gpxx`.

Such gpx file is still valid, but parser will not be able to read color:
```
<?xml version="1.0" encoding="UTF-8"?>
<gpx version="1.1"  
  xmlns="http://www.topografix.com/GPX/1/1" 
  xmlns:garminEtenstions="http://www.garmin.com/xmlschemas/GpxExtensions/v3">
  <trk>
    <name>Track name</name>
        
    <extensions>
      <garminEtenstions:DisplayColor>red</garminEtenstions:DisplayColor>        
    </extensions>
        
    <trkseg>
      <trkpt lat="47.4191933" lon="8.5075483">
        <time>2016-07-19T17:10:34</time>
      </trkpt>
    </trkseg>
  </trk>
</gpx>
```

Reply:
I have to discard the namespace. Seems it depends of the source device generates the gpx file. My fault.
But you sample is bizar, you should have the tree as follow:
```
  <trk>
    <name>3-DeviationTT-1</name>
    <extensions>
      <TrackExtension>
        <DisplayColor>Cyan</DisplayColor>
      </TrackExtension>
    </extensions>
```

Reply:
It needs to read gpx specs from garmin ...